### DISCUSSION TITLE
Settings Sync

### DISCUSSION THREAD

**Discussion by JosephTLyons** [zed-industries/zed#6569] [Feature Requests]
We've had a few issues come up asking us to add a settings sync feature to Zed:
- https://github.com/zed-industries/community/issues/952
- https://github.com/zed-industries/community/issues/265

While this is not currently on our roadmap, and it will likely not be added anytime soon, I'm interested to hear what all users would want from this sort of a feature. 

IMO, we'd be able to sync:

- `settings.json`
- `keymap.json`
- user's installed extensions

Blockers:

- Keymaps are not currently portable. We need a way to allow users to define a single keymap that works on all operating systems.
- We need maintain a list of extensions installed so they can be installed again on another system, programmatically

What all would you want synced and restored between different computers running Zed?

**Comment by JosephTLyons** [comment]
VS Code's setting sync has worked out pretty well for me, really smooth, completely forget that it's there, but I never liked that my settings were stored somewhere privately; I liked my settings being stored in a location I had control over.  That might be a bit biased since I came from Atom prior and having to sync my settings with a GitHub Gist.

**Reply by zongUMR** [reply]
Maybe you can refer to sublime_text's sync strategy, it has a  third-party package can sync the all settings to a private github gist, https://packagecontrol.io/packages/Sync%20Settings

**Reply by calebmeyer** [reply]
I was so disappointed the first time I tried to share my list of extensions with someone from VS Code. Yes, it's really smooth syncing, but the gist that both atom/sublime use is human readable and scriptable.

**Comment by calebmeyer** [comment]
What I'd like to see:

Sync Settings has four commands: backup to remote, restore from remote, create profile, and switch profile. They all look at a git repository in any code host (github/gitlab/bitbucket, etc)

Sync Settings prompts you for that repository's URL on first launch, and automatically gets/sets up an access token if needed (if the repo is private). Then it downloads that repo and uses it to set up the editor.

Inside the repo is plaintext settings files in folders. Each folder is a complete configuration setup (settings + keybinds + extensions/packages + whatever else) and the folder name is the name of a profile (e.g. work, writing, rust, etc.). If exactly one folder exists, it is treated as the default for all operations. The default is otherwise the folder named `default` or the first folder alphabetically. 

If you have multiple profiles, you can relaunch the editor with a different profile using the switch profile command.

**Comment by photex** [comment]
I really like the "settings repository" plugin for CLion. I point it to a git repo and it does the rest.

I appreciate the simplicity of the VScode sync, but it's also a bit shady isn't it. (Yes, I'm aware of the JetBrains sync plugin that works similarly to VSCode as well).

The reason I prefer settings sync that's sort of built-in vs a third party is two parts, the first part is trust, the second part is support. 

By trust I mean, I'm perhaps a paying customer of Zed and I can trust that the feature is implemented by the company and I'm trusting a single entity in this situation (I don't have to worry about a rogue update to the extension that starts to siphon off secrets from my computer).

By support I mean, since it's a tier 1 product feature and so will support sync'ing all relevant settings, extensions, and themes. I don't have to worry about it getting abandoned (like I worry about most of the Sublime ecosystem).

**Comment by rcrooks** [comment]
I would be happy even to be able to change the location of the settings json file to a shared folder (dropbox or google drive or apple cloud or whatever) to be able to keep the settings in sync between my two machines. If we adopted this for my team at work, it would be great to have the option of sharing settings with the whole team (with the choice of using the team shared settings or not).

**Reply by DaveEveritt** [reply]
+1 for this. Leaves the settings file under user control, but in a space shared between machines. However, each cloud service has its own protocols and quirks, so some restrictions as to which services are supported is likely.

**Comment by mattly** [comment]
is there any reason why settings sync couldn't happen via source control, like other dotfiles?

**Reply by mattly** [reply]
since posting this 10 months ago, Zed has done two things that have made it more difficult:

1. Littered `.config/zed` with a bunch of `.tmpxxxxx` files, which *can* be gitignored but ultimately make working with that directory in Zed itself much less pleasant

2. Added extensions, the manifest or state of which are kept in a different directory. I would very much love to have this stored in a version-controlled data file. I'd gone to a bit of trouble to create this ability for myself with vscode:
    - https://github.com/mattly/dotfiles/blob/master/bin/codex
    - https://github.com/mattly/dotfiles/blob/master/configs/.config/vscode/extensions
  
    But with Zed it requires some workarounds right now, such as symlinking `~/Library/Application Support/Zed/extensions/manifest.json` and possibly other things, but I'm not sure it would rebuild everything else based on a manifest.

I feel very strongly that my editor setup should be able to be managed and synced through version control. In *addition* to the benefits of version control, it also allows for better partitioning between usage contexts, which is why VSCode settings sync doesn't work for me. 

I have created per-employer or per-client github accounts when I am assigned a machine owned by that company, because I do *not* sign into personal accounts on machines I do not own. With my settings stored in git, I can very easily share the repository between those accounts, sync changes back and forth and such. I can't do that with VSCode Settings Sync because the sync is tied to a single github account, which would break the usage context barriers I have setup.

Making it easy to manage settings via source control would *also* mean less infrastructure to build around Zed. The "Sign In" feature currently requires a Github account, but how much of the editor do you want to base around people having github accounts?

**Reply by kergoth** [reply]
It's not perfect, as it doesn't sync removal, but the `auto_install_extensions` setting helps with this.

**Reply by erhhung** [reply]
> Littered .config/zed with a bunch of `.tmpxxxxx` files, which can be gitignored but ultimately make working with that directory in Zed itself much less pleasant

Why are these `.tmpxxxxx` files under `$XDG_CONFIG_HOME` (`~/.config/zed`) in the first place?
Isn't `$XDG_STATE_HOME` (`~/.local/state/zed`) or `$XDG_CACHE_HOME` (`~/.cache/zed`) the more appropriate locations?

**Comment by skyf0cker** [comment]
settings and keymap.
When I changed a keymap on my work Mac, I needed to sync it with my home Mac after work. It's not difficult, but it's annoying.

**Reply by enkr1** [reply]
can you kindly show ur workaround please?

**Comment by enkr1** [comment]
whats the latest update in syncing your settings, extensions, etc across multiple devices right now?

Thank you!

**Reply by enkr1** [reply]
really wish for an update man ._.

**Comment by Aur0nd** [comment]
I'd really love to see zed smashing but lacking little features such as this makes me go back to intelli...

**Comment by FlorianWendelborn** [comment]
What about manually syncing `~/.config/zed`? Is this the currently intended way? Is there any non-obvious downsides to this that I’m missing? I’m also interested in settings sync, but if officially encouraged I could easily live with syncing `~/.config/zed`

**Comment by Jervx** [comment]
Hell yeah, settings sync will save us from doing custom settings over and over again

**Comment by aretrace** [comment]
<img width="462" alt="image" src="https://github.com/user-attachments/assets/99b0a19d-824f-4113-951a-e5e53a655729">

**Comment by jonathanjsimon** [comment]
Just adding my voice to the desire for settings sync :-)

**Comment by lougreenwood** [comment]
I sync my dotfiles, no need for settings sync.

**Comment by devdumpling** [comment]
+1 the need, coming from emacs and VS Code, but for me personally it'd be enough to just have a toggleable auto-push-on-save (debounced for things like theme switching) of my settings to a repo I select. That's not something Zed necessarily needs to do either, but could be a neat quality of life thing.

Like many devs, I'm very particular about my setup and I work across a few machines. I have a "set up all my shit" repo for init'ing on new machines specifically to help with this. Currently just manually pushing my config to that repo 🤷

**Reply by vraravam** [reply]
> Like many devs, I'm very particular about my setup and I work across a few machines. I have a "set up all my shit" repo for init'ing on new machines specifically to help with this. Currently just manually pushing my config to that repo 🤷

100% - this is what I do as well. Would love to see sync mechanism in-built into zed so that when I go to a new machine, it's all ready (including extension installation and configuration)

**Comment by wojtyniak** [comment]
Can we at least get anything that is not a static configuration removed from `~/.config/zed`? It would make syncing the folder much easier.

**Reply by nikaro** [reply]
Kind of related: https://github.com/zed-industries/zed/discussions/37097

**Comment by devdumpling** [comment]
For the time being I'm "syncing" my `~/.config/zed` into a private repo I use to sync configs for a number of apps across multiple machines (and for making initial setup on new machines easier). Pretty straightforward and haven't had any issues. It's not quite the same as having it built into the editor but works for me. I'm not using any extensions or anything yet though so not sure if it would cover that or other use-cases. 

I was able to get what I need (sync keymaps, editor settings, AI conversations, and basically everything in `.config/zed` that isn't a tmp file with a simple script that runs on a cron or manually if needed. It uses `rsync` to omit the `tmp*` files. 

```sh
rsync -av --exclude='.tmp*' "$ZED_SOURCE_DIR/" "$ZED_DEST_DIR/"
```

Here's a gist with the script I use in case it's helpful to anyone:

https://gist.github.com/devdumpling/206cddd544be229ef27af50ce7f80c29

**Comment by rome-user** [comment]
I use home-manager in Nix to manage the all configuration of a computer in a single file. Home-manager supports various editors (Emacs, VS Code, etc.). The ideal option for Nix users is exposing all settings in `settings.json`. 

I currently use home-manager to effortlessly recreate desktop and development environments across several computers (this works on Mac and Linux).

**Comment by auno** [comment]
One issue that I just noticed when going for the standard solution of syncing settings through a git repository is that it clashes with the Remote Development feature in some cases. This happens because SSH connections, which might be different for different hosts, are stored in the regular settings file.

In my case, I use two separate workstations, one at home and and one at work, which both should be able to remote into a third computer, my work server. The settings for doing so are different for the two workstations though, so syncing the settings between them in that way would ensure that one is always broken. I also remote into my home workstation from my work one, but not the other way around, which further adds to the difference between those settings files.

This might be a corner case when it comes to the use of the Remote Development feature, but I would guess that I'm not the only one it applies to. The (conceptually) simple solution to this issue would of course be to separate out the SSH connections to their own configuration file, that one could choose to sync or not depending on ones needs.

**Reply by FlorianWendelborn** [reply]
Did you try using `~/.ssh/config` to configure the different settings for those hosts? I don't use this with Zed, but it's supported for quite a few tools and workflows using SSH, so it might be worth checking out if it works for you

**Comment by DaveEveritt** [comment]
Even copying a single file from one machine to another is a tedious way to sync settings, but at least a clear guide about this in lieu of the actual feature would help a lot. Or allow us to *set the location of settings files* that we can access from other machines? If they’re plain text, and owned/write only by the user what would be the issue?

**Comment by hedefalk** [comment]
I'm using a dotfiles repo with gnu stow, but now moved to nix mkOutOfStoreSymLink for just settings.json and keymap.json and it mostly works fine, but things like  "buffer_font_size" is really annoying because I have different DPI's on different machines and also on different screens within the same machine. So I really feel strongly that shouldn't be in the same scope of settings but rather something within the window.

**Comment by JustyleR** [comment]
still waiting

**Reply by MrBns** [reply]
Wait forever. the reason zed team is avoiding this because of those people who put 👎🏻  to  your comments.

**Comment by SinarPandora** [comment]
I hope we can have the setting sync, so that I can make Zed my main editor everywhere.
I really love Zed's performance and the clear UI.

**Comment by DanteMarshal** [comment]
Suggestion, instead of implementing a full-fledged automatic sync system we can start with a simple manual sync that takes all the necessary files from / into a git repository, and does it manually using simple "push settings" and "pull settings" commands in the editor.
The user will have control over when to push or pull, and should be able to view the status of their changed settings just like in a git repo.
I guess what I'm asking is just a special UI overlay designed specifically for managing settings and keybindings sync (ideally with support for profiles) over git in the editor, so we don't have to setup the git repo and manage the linking manually, which eases the workflow.

What to sync, my suggestion is Settings + Keybindings + Extensions (list of installed extensions + their settings). It'd be ideal if we can have profiles each containing a set of these and even more perfect if profiles can "inherit" from each other.
Like, have a main profile for settings and keybindings, then have different profiles that extend that and only modify the installed and active extension-sets for e.g. different tech-stack setups.

**Reply by vittorius** [reply]
I'm currently working on an extension that does the "push" part, although somewhat limited due to the limited capabilities of Zed extensions in general. Stay tuned 🙂

**Reply by vraravam** [reply]
eagerly awaiting your extension @vittorius !

**Reply by vittorius** [reply]
Just created a PR into the Zed extensions repo with the first stable version of it https://github.com/zed-industries/extensions/pull/4557. I have been using it for a long time in the form of the [dev extension](https://github.com/vittorius/zed-settings-sync?tab=readme-ov-file#dev-extension-installation), actually. Works fine. You can start using it in the same way at the moment.

**Comment by christianwgd** [comment]
Here's what I have done to sync the settings between my two Macs: Moved the "zed" directory to my personal Nextcloud and versioned it there with Git / Github. After that I've "sum-linked" (ln -s <sourcedir> <targedir>) the directory to the "~/.config" directory of my two Macs. 
Seems at least an easy workaround.

**Reply by tibitTW** [reply]
can you sync the installed extension? I use dotfiles and stow to sync my config

**Reply by Malix-Labs** [reply]
I use Nix Home Manager too but I'd like something built-in with a GitHub/GitLab account

**Reply by Daste745** [reply]
> can you sync the installed extension? I use dotfiles and stow to sync my config

You can use the `auto_install_extensions` setting, for example:
```json
"auto_install_extensions": {
  "colored-zed-icons-theme": true,
  "git-firefly": true,
  "dockerfile": true
}
```
Get these keys via extensions -> ... -> Copy Extension ID @tibitTW

**Comment by labadcloyd** [comment]
I'm leaving my comment here to express the same desire for this feature to be implemented 😊

**Comment by PolpOnline** [comment]
Also I'd like to point out that #26043 probably needs to be implemented beforehand, because I wouldn't trust syncing my secrets (MCP API tokens and such) in a repo I have control of, let alone someone else's server

**Comment by mauriciord** [comment]
There's been a fundamental limitation: Zed currently does not provide direct APIs to access or modify editor settings, keymaps, or list installed extensions. Zed's extension system operates in a sandboxed WebAssembly environment with minimal access.

**Reply by Malix-Labs** [reply]
> Zed currently does not provide direct APIs to access or modify editor settings, keymaps, or list installed extensions

They are just JSON dotfiles in `~/.config/zed`, `~/Library/Application Support/Zed/extensions`, `~/.local/share/zed/extensions`, `$XDG_DATA_HOME/zed/extensions`

**Reply by mauriciord** [reply]
@Malix-Labs 

> > Zed currently does not provide direct APIs to access or modify editor settings, keymaps, or list installed extensions
> 
> They are just JSON dotfiles in `~/.config/zed`

That's the point, explain how to list installed extensions using `~/.config/zed`

**Reply by Malix-Labs** [reply]
@mauriciord : https://zed.dev/docs/extensions/installing-extensions#installation-location

you can then use jq, nushell, serde, ... to parse the JSON files

I don't know if extensions has access to such tools if that's what you're asking

**Reply by Malix-Labs** [reply]
See #56088

**Comment by jorhett** [comment]
Who knew? This feature already exists in Gemini's addled mind

<img width="697" height="330" alt="Screenshot 2025-11-13 at 9 59 51 AM" src="https://github.com/user-attachments/assets/905729aa-4a8d-4615-bd3f-cd8f1d529dda" />

**Reply by Malix-Labs** [reply]
> Zed, the code editor, offers a built-in "Settings Sync" feature

> You can use any Git hosting service (e.g., GitHub, GitLab, Bitbucket) to manage your synced settings.

<img width="405" height="213" alt="meme: well no, but actually yes" src="https://github.com/user-attachments/assets/e8a0340c-23f8-4696-8edf-81611e93b02e" />

**Reply by franciskafyi** [reply]
ah phew, that was easy to "implement" 😆

**Comment by MrBns** [comment]
Is there any Way to sync Installed Extension yet? its kinda boring to install All extension manually.

**Reply by Daste745** [reply]
Assuming you have *a way* of syncing the settings.json file already, you can use the solution from [my previous comment](https://github.com/zed-industries/zed/discussions/6569#discussioncomment-13743071):
```json
"auto_install_extensions": {
  "colored-zed-icons-theme": true,
  "git-firefly": true,
  "dockerfile": true
}
```

**Reply by MrBns** [reply]
> Assuming you have _a way_ of syncing the settings.json file already, you can use the solution from [my previous comment](https://github.com/zed-industries/zed/discussions/6569#discussioncomment-13743071):
> 
> ```json
> "auto_install_extensions": {
>   "colored-zed-icons-theme": true,
>   "git-firefly": true,
>   "dockerfile": true
> }
> ```

Okay honestly, Zed is really messing with one single settings file. atleast separate file for extension. 

and also for remote. because I don't wanna sync my remote config.

**Reply by Malix-Labs** [reply]
See #56088

**Comment by shssoichiro** [comment]
> Keymaps are not currently portable. We need a way to allow users to define a single keymap that works on all operating systems.

Regarding this issue, there aren't a lot of keys that differ between PC and Mac, and even fewer between Win and Linux.

A couple of ideas:
- Have the ability to specify overrides for an OS, such as this example:
```
"bindings": {
  "ctrl-tab": "pane::ActivateNextItem",
  "ctrl-shift-tab": "pane::ActivatePreviousItem",
  "Mac": {
    "cmd-tab": "pane::ActivateNextItem",
    "cmd-shift-tab": "pane::ActivatePreviousItem",
  }
},
```
In this example, `ctrl+tab` and `ctrl+shift+tab` would be used, except on Mac, where `cmd+tab` and `cmd+shift+tab` would be used instead.
- Interpret similar keybindings according to the currently active OS. This would match the way OS-level keymaps work, for example "Cmd/Win/Super" would all refer to the same key, and "Alt/Option" would refer to the same key.

For me, being able to sync settings, keymaps, extensions, and theme settings would be the biggest missing feature between Zed and VSCode. Zed already has online accounts integrated into the editor, which is typically the larger part of the work. I very much hope we can get this sorted out in 2026, 3 years after the opening of this discussion. I know that I currently can manually sync the files using a variety of methods, but it's annoying, so I would much rather have an automated solution.

**Reply by skhaz** [reply]
Give up, they do not care. I am happy now with neovim.

**Reply by shssoichiro** [reply]
@skhaz This type of comment is completely useless. Please be productive.

**Reply by skhaz** [reply]
Yes! I am productive with neovim. 

Zed have another priorities now. Wake up.

**Comment by vittorius** [comment]
Shout to all following this thread: I've created a pretty simple extension for it (based on a custom LSP "hack") and just opened a PR to the extensions repo that adds it: https://github.com/zed-industries/extensions/pull/4557
The extension itself is here: https://github.com/vittorius/zed-settings-sync. Please comment, star, and support this contribution to Zed extensions if you find it useful.

**Comment by VihangaFTW** [comment]
"vscode killer" 
-> No jupyter notebook support
-> no setting sync

**Reply by MrBns** [reply]
you just got disliked to tell the truth. 
they missed the basic frequent feature, "settings" sync. 

even you see they didn't list the `Extension Sync` in the list.

**Comment by ssanumand** [comment]
Please bring support for settings sync, please...

**Comment by tianqiii** [comment]
I use chezmoi to control my private dotfiles, so if there has some way to store extensions' indexes in settings.json or somewhere else, and to install them on different machines automatically, is ok for me.

**Reply by Malix-Labs** [reply]
I also use nix's home-manager symlinks, but a built-in settings sync a la vscode would make for a good QoL enhancement

**Reply by injust** [reply]
> store extensions' indexes in settings.json or somewhere else, and to install them on different machines automatically

@tianqiii If you're trying to auto-install extensions, you can do that with the `auto_install_extensions` setting.

**Reply by azadevlab** [reply]
I use chezmoi too :)
easy peasy lemon squeezy
ofc if you extensions list is static, you can just manually grab it once and manually add in settings.json (`jq -r '.extensions | keys | map({(.): true}) | add' ~/.local/share/zed/extensions/index.json`)
but due we're using chezmoi, let's do it in the right way 

```
~/.local/share/chezmoi/home/dot_config/zed/settings.json.tmpl
{{- $indexPath := joinPath .chezmoi.homeDir ".local/share/zed/extensions/index.json" -}}
{
  "auto_install_extensions": {
    {{- if stat $indexPath }}
      {{- $index := (include $indexPath | fromJson) -}}
      {{- $exts := keys $index.extensions -}}
      {{- range $i, $name := $exts -}}
        "{{ $name }}": true{{ if lt $i (sub (len $exts) 1) }},{{ end }}
      {{- end }}
    {{- else }}
      "html": true,
      "dockerfile": true
    {{- end }}
  },
  "telemetry": {
    "diagnostics": false,
    "metrics": false
  },
    "session": {
    "trust_all_worktrees": true
  },
  "icon_theme": {
    "mode": "system",
    "light": "Zed (Default)",
    "dark": "Zed (Default)"
  },
  "base_keymap": "Atom",
  "ui_font_size": 16,
  "buffer_font_size": 15,
  "theme": {
    "mode": "system",
    "light": "One Light",
    "dark": "One Dark",
  },
}
```

**Reply by Malix-Labs** [reply]
Also see #56088

**Comment by marcos2872** [comment]
@JosephTLyons 


**Feature Implementation: GitHub Settings Sync (Private Repo + PAT)**

I've built a complete `settings_sync` crate that syncs `settings.json`, `keymap.json`, and `tasks.json` to a **private GitHub repo** using a Fine-grained PAT:

**Key Features:**
- Push/pull via Command Palette (`Sync Settings To Git` / `Pull Settings From Git`)
- Token stored securely in OS keychain (never on disk)
- **Private repos only** (public repos rejected for security)
- Automatic `pull --rebase` on push conflicts
- `git reset --hard` on pull (remote always wins, no merge hell)
- Full error handling with actionable toasts
- Concurrent operation protection

**Demo**: https://github.com/marcos2872/zed-contributing/pull/1

**How it works** (setup in 60s):
1. Private GitHub repo + Fine-grained PAT (`Contents: Read/Write`)
2. Add `"sync.git.repo_url"` to `settings.json`  
3. `Cmd+Shift+P` → `Set Settings Sync Token`
4. `Cmd+Shift+P` → `Sync Settings To Git`

Would love maintainer feedback! Happy to iterate and target `zed-industries/zed` if directionally correct.

**Reply by vittorius** [reply]
Hello my fellow-Zed-syncer 🙂 and welcome https://github.com/vittorius/zed-settings-sync. Unfortunately, the Zed team [won't likely consider](https://github.com/zed-industries/extensions/pull/4557) such an approach aligned with their vision.

**Comment by MrBns** [comment]
## its been 2 years.

Its really a Bad thing. This Settings Sync discussion been for 2 years now. Many people made their Own way syncing. I do sync using git repo. but it only works for Settings.json and Keymap.json. 

what about installed extensions, theme, and others things that need to be sync. at least zed team should store installed info in ~/.config/zed like `settings.json`.  Its kinda Shame too after 2 years No solution for this simple problem.

**Reply by dany-on-demand** [reply]
I feel cheated, what the fuck did I create an account for if not for settings sync?

**Reply by Malix-Labs** [reply]
[`auto_install_extensions`](https://zed.dev/docs/reference/all-settings#auto-install-extensions) and [`auto_update_extensions`](https://github.com/search?q=repo%3Azed-industries%2Fzed%20auto_update_extensions&type=code)

Also see `~/.local/share/zed/extensions`, and especially `~/.local/share/zed/extensions/index.json` and `~/.local/share/zed/extensions/installed`

**Reply by MrBns** [reply]
> [`auto_install_extensions`](https://zed.dev/docs/reference/all-settings#auto-install-extensions) and [`auto_update_extensions`](https://github.com/search?q=repo%3Azed-industries%2Fzed%20auto_update_extensions&type=code)
> 
> Also see `~/.local/share/zed/extensions`, and especially `~/.local/share/zed/extensions/index.json` and `~/.local/share/zed/extensions/installed`

what do you think ? is this the best solution ? whats wrong with storing installed extension on same config folder. I don't understand whats the reason behind keeping their folders separate.

**Reply by MrBns** [reply]
> I feel cheated, what the fuck did I create an account for if not for settings sync?

its for using their AI services. not for setting sync. LOL

**Reply by Malix-Labs** [reply]
https://github.com/zed-industries/zed/discussions/6569#discussioncomment-16839438 from @MrBns :
> what do you think ? is this the best solution ? whats wrong with storing installed extension on same config folder. I don't understand whats the reason behind keeping their folders separate.

I think it is wrong indeed

I just created #56052 about it

**Reply by injust** [reply]
> I think `~/.local/share/zed/extensions/index.json` should be moved to `~/.config/zed/extensions.json`

No, it shouldn't. It's not a config file.

**Reply by Malix-Labs** [reply]
_let's continue this discussion in #56052_

**Reply by Malix-Labs** [reply]
See #56088

**Comment by Malix-Labs** [comment]
# Workaround

## Symlinking using Nix Home Manager

```nix
{
  config,
  ...
}:
{
	xdg.configFile = {
		"zed".source = config.lib.file.mkOutOfStoreSymlink "{config.home.homeDirectory}/path/of/your/choice/zed";
	};
}
```

**Reply by valdoryu** [reply]
Why no symlinking the file instead of the folder ?

**Reply by Malix-Labs** [reply]
@valdoryu because there are multiple files and folders in this folder

**Comment by timephy** [comment]
I want to switch to Zed too, having to set Zed up on the 2 machines I work on regularly makes me a bit dissatisfied

**Reply by MrBns** [reply]
wait one more decade. after all Complex Shitty AI stuff is completed they might think about settings sync. 😂

**Reply by valdoryu** [reply]
Well symlinking works pretty good

**Comment by tianqiii** [comment]
LOL, I switch back to neovim eventually.😂

**Reply by skhaz** [reply]
neovim + tmux + ghostty best combo ever!

**Reply by Malix-Labs** [reply]
This is unrelated as NeoVim also does not have native settings sync and requires, like Zed, to use dotfiles configuration generators (i.e. Nix Home Manager) or symlink dotfiles manager (i.e. GNU Stow)

**Reply by MrBns** [reply]
> This is unrelated as NeoVim also does not have native settings sync and requires, like Zed, to use dotfiles configuration generators (i.e. Nix Home Manager) or symlink dotfiles manager (i.e. GNU Stow)

but they do have a way to manage all syncing stuff at one place.

**Comment by FlyinPancake** [comment]
it would be great if zed supported multiple settings files like `settings.ssh.json` `settings.local.json` `settings.whatever.json`

naturally merging these would be inconveinient but currently zed puts EVERYTHING in settings.json like remote ssh histroy, which is the wrong place for it

**Reply by anilzeybek** [reply]
see: https://github.com/zed-industries/zed/discussions/23663

**Comment by zepc007** [comment]
It's still not implemented till 2026. It's so embarrassing.

**Comment by mhawash** [comment]
I have just switched away from VSCode to Zed recently, and after spending some time learning the differences I find out there is no settings sync! It shouldn't be that complicated to add this feature.

**Reply by MrBns** [reply]
main problem is they deosn't store anything in a central place. atleast the settings. they store settings in config dir and extenstions related stuff in another place. so yeah its the problem here.

**Reply by valdoryu** [reply]
Pretty sure they could use github account storage for multiple files, right  ?

**Reply by ReFleXzZ** [reply]
> main problem is they deosn't store anything in a central place. atleast the settings. they store settings in config dir and extenstions related stuff in another place. so yeah its the problem here. 

That is not even a problem, because you can just sync settings to different directories. That's not pretty much. Zed has files in .local/state and some in .config, as the XDG Desktop specification requires it. Doesn't mean it's a problem to sync between different devices

**Comment by osawereao** [comment]
QUESTION: is the feature available yet?
It is really important and I would say essential to some engineers, including me

**Comment by kynrai** [comment]
Just switched to zed full time and installed on a second machine only to learn this is not a thing, 

Would me just using syncthing or manually setting up a dot config sync work? how are extensions etc managed? im un sure

**Reply by valdoryu** [reply]
You can backup your config through a github repo and use a symlink logic to keep it in sync

However extensions are much more annoying to sync as you will have to sync their node_modules folder as well

**Comment by 1antares1** [comment]
Does anyone know at least one manual way to sync that's suitable for 2026?

Because if it requires a sign-up, they must be saving our data somewhere... I definitely use Zed for its performance compared to Cursor, but that topic has been going on for 3 years and nothing has happened.

**Reply by enkr1** [reply]
yes bro just give up HAHAAHAHAHAHAHA

**Reply by timephy** [reply]
😂

**Reply by vittorius** [reply]
The VS Code fork idea is outdated; we must do a Zed fork! With settings sync and all other shiny stuff baked in.

**Comment by skhaz** [comment]
> I definitely use Zed for its performance compared to Cursor

Then use neovim.

**Comment by axellpadilla** [comment]
Pretty important feature...

**Comment by macraig** [comment]
Opened an issue to track this here: https://github.com/zed-industries/zed/issues/59978

**Comment by twinkle311** [comment]
We have GitHub-based login functionality, so why not build a personal configuration sync mechanism based on GitHub's private repos?

**Reply by Alistair1231** [reply]
even just a gist with settings and keymap json files

**Comment by markfaine** [comment]
What's the point of having a GitHub login  if settings aren't synced? That's the only reason I can think of for logging in at all.

**Reply by alshdavid** [reply]
I literally thought that's why I was logging in

**Comment by Madi-S** [comment]
bump

**Comment by Ferrb9579** [comment]
I need this so that I can keep my Windows and Linux machine in sync

**Comment by rubiin** [comment]
+1

**Comment by VihangaFTW** [comment]
yall still using zed🤣?

**Comment by cig0** [comment]
I'd pretty much welcome a syncing feature that keeps all my (three) Zed instances synced (across macOS and GNU/Linux hosts), including Zed settings, extensions, themes, and so on.

Because different hosts have different specifications, e.g., display resolution or GPU muscle to run local AI models, having a way to override settings by host (I'm thinking about Kustomize) would be ideal 👌

**Reply by berkus** [reply]
Yes, please! I also have a fleet (more than 4) of Linux and macOS machines with slightly different Zed configs, right now it's a bit tedious to propagate config manually into a gist, then copy paste the diff into other machines' configs.

Having per-machine settings in the synced config would be awesome.

**Comment by alshdavid** [comment]
Bump

**Comment by Leon135** [comment]
bump

**Comment by byuwur** [comment]
The fact that this isn't a thing after THREE YEARS just explains why I can take this editor seriously.

**Comment by byuwur** [comment]
The fact that this isn't a thing after THREE YEARS explains why I can't take this project seriously.

**Comment by byuwur** [comment]
The fact that this isn't a thing after three years explains why I can't take this project seriously.

**Reply by vraravam** [reply]
as of this morning, I have moved back to VSCodium for this exact reason. good bye.
